### ISSUE TITLE
MD034 False positive with reference links in blockquote

### ISSUE THREAD

**Issue by mraber** [rvben/rumdl#674]
### rumdl version

0.2.17

### Installation method

Homebrew

### Bug description

`rumdl` detects MD034 "bare URL" when using a reference link inside a blockquote (admonitions).

### Steps to reproduce

```bash
cat<<_EOF | rumdl check --stdin
# Test MD034 within blockquote

[A link] in paragraph is OK.

> But [A ref link] in blockquote is not unless defined elsewhere ([A link]).
> That bare link <https://www.foo.com\> works.
> [A ref link]: https://www.google.com

[A link]: https://www.github-01.com
_EOF
<stdin>:7:17: [MD034] URL without angle brackets or link formatting: 'https://www.google.com' [*]

Found 1 issue(s) in <stdin>
```

### Expected behavior

No MD034 when a reference link is declared in a blockquote.

### Actual behavior

If a reference link is declared in a blockquote, a MD034 error is raised.

### Configuration

```toml

```

### Operating system

macOS

**Comment by rvben** [comment]
Thanks for the report, @mraber! This is fixed in v0.2.19.

Root cause: rumdl's reference-definition parser only matched definitions at the start of a line, so a `> [ref]: url` definition inside a blockquote was invisible to it. MD034 then treated the URL as a bare URL. The parser now recognizes blockquote-prefixed reference definitions (they're document-global per CommonMark), so `> [A ref link]: https://www.google.com` is no longer flagged. MD052's "reference not found" false positive for the same case is fixed too.

`cargo install rumdl` (or `cargo binstall rumdl` for a prebuilt binary) to upgrade. Closing, but please reopen if you still hit it.

**Comment by mraber** [comment]
@rvben Thanks a lot! 👍
